inference-engine/src/mkldnn_plugin/memory_descs/cpu_memory_desc_utils.cpp

```diff
@@ -0,0 +1,113 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "cpu_memory_desc.h"
+#include "memory_descs/cpu_memory_desc_utils.h"
+#include "mkldnn_memory.h"
+#include "memory_descs/dnnl_blocked_memory_desc.h"
+#include "utils/general_utils.h"
+#include "utils/cpu_utils.hpp"
+#include <limits>
+#include <vector>
+#include <numeric>
+#include <blob_factory.hpp>
+#include <dnnl_types.h>
+
+using namespace mkldnn;
+using namespace MKLDNNPlugin;
+using namespace InferenceEngine;
+
+namespace MKLDNNPlugin {
+
+std::unique_ptr<DnnlMemoryDesc> MemoryDescUtils::convertToDnnlMemoryDesc(const MemoryDesc& desc) {
+    if (MemoryDescType::Blocked == desc.getType()) {
+        return convertToDnnlMemoryDesc(*(desc.as<CpuBlockedMemoryDesc>()));
+    } else if (MemoryDescType::Mkldnn & desc.getType()) {
+        return std::unique_ptr<DnnlMemoryDesc>(dynamic_cast<DnnlMemoryDesc *>(desc.clone().release()));
+    } else {
+        IE_THROW() << "Cannot convert MemoryDesc to DnnlMemoryDesc";
+    }
+}
+
+std::unique_ptr<DnnlMemoryDesc> MemoryDescUtils::convertToDnnlMemoryDesc(const CpuBlockedMemoryDesc& desc) {
+    return std::unique_ptr<DnnlBlockedMemoryDesc>(new DnnlBlockedMemoryDesc(desc.getPrecision(), desc.getShape(), desc.getBlockDims(),
+                                                              desc.getOrder(), desc.getOffsetPadding(), desc.getOffsetPaddingToData(), desc.getStrides()));
+}
+
+std::unique_ptr<DnnlBlockedMemoryDesc> MemoryDescUtils::convertToDnnlBlockedMemoryDesc(const InferenceEngine::TensorDesc& desc) {
+    const auto &blkDesc = desc.getBlockingDesc();
+    return std::unique_ptr<DnnlBlockedMemoryDesc>(new DnnlBlockedMemoryDesc(desc.getPrecision(), Shape(desc.getDims()), blkDesc.getBlockDims(),
+                                                                                blkDesc.getOrder(), blkDesc.getOffsetPadding(),
+                                                                                blkDesc.getOffsetPaddingToData(), blkDesc.getStrides()));
+}
+
+MemoryDescPtr MemoryDescUtils::applyUndefinedOffset(const MemoryDesc& desc) {
```

Comment:
M.b. rename the function to emphasize what it really does, i.e. not only undefine the offset but also undefine strides.